src/local.rs

```diff
@@ -15,21 +15,36 @@ impl<'sc, T> Clone for Local<'sc, T> {
 }
 
 impl<'sc, T> Local<'sc, T> {
+  /// Creates a new empty local handle.
+  pub fn empty() -> Self {
+    Self(ptr::null_mut(), PhantomData)
+  }
+
   pub unsafe fn from_raw(ptr: *mut T) -> Option<Self> {
-    Some(Self(NonNull::new(ptr)?, PhantomData))
+    Some(Self(ptr, PhantomData))
+  }
+
+  /// Returns true if the handle is empty.
+  pub fn is_empty(&self) -> bool {
+    self.0.is_null()
+  }
+
+  /// Sets the handle to be empty. IsEmpty() will then return true.
+  pub fn clear(&mut self) {
+    self.0 = ptr::null_mut();
   }
 }
 
 impl<'sc, T> Deref for Local<'sc, T> {
```

Comment:
I suggest that since Locals can hold nullptr we do not implement Deref but instead implement something like this
https://doc.rust-lang.org/std/primitive.pointer.html#method.as_ref 
```rust
impl<'sc, T> Local<'sc, T> {
  unsafe fn as_ref<'a>(self) -> Option<&'a T> {
    if let Some(ref) = unsafe { self.0.as_ref() } {
      Some(ref)
    } else {
      None
    }
  }
}
```
What do you think Bert?

Reply:
But in many cases Locals can't hold nullptrs. Basically every time a Local is returned V8 will use a MaybeLocal if it is potentially empty.
Some V8 APIs do accept empty locals as parameters. But I think we can just model those as Option<Local>'s as well.